data_collection/gazette/spiders/pa/pa_parauapebas.py

```diff
@@ -0,0 +1,47 @@
+import datetime as dt
+
+import scrapy
+from dateparser import parse
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class PaParauapebasSpider(BaseGazetteSpider):
+    TERRITORY_ID = "1505536"
+    name = "pa_parauapebas"
+    start_urls = ["http://apps.ioepa.com.br/Parauapebas/Busca"]
+    start_date = dt.date(2022, 1, 4)
+
+    def parse(self, response):
+        ano_atual = dt.datetime.now().year
+        for year in [2022, ano_atual]:
+            for month in range(1, 13):
+                for day in range(1, 32):
+                    if len(str(day)) < 2:
+                        day = "".join(("0", str(day)))
+                    if len(str(month)) < 2:
+                        month = "".join(("0", str(month)))
+                    data = f"{year}-{month}-{day}"
+                    url = f"{response.url}?data={data}"
+                    yield scrapy.Request(
+                        url,
+                        callback=self.parse_gazette,
+                        cb_kwargs={"data": data},
+                    )
+
+    def parse_gazette(self, response, data):
+        link_diario = response.css("a[target]").css("::attr(href)").extract_first()
+        data_do_diario = link_diario.split("/")[-1].split(".")[:-2]
```

Comment:
Alguns comentários aqui, em tom de sugestão:
1. Não precisaria encadear uso do seletor, fica um pouco menor colocar tudo na mesma chamada do `.css`
2. `.extract_first` está em desuso, dê preferência a usar `.get`
3. Porém, ao deixar de usar `.get` acima podemos aproveitar o método `.re_first` da resposta do seletor e deixar a construção da variável da data mais legível
4. Como já temos a data da publicação no argumento `data`, mudaria o nome dessa aqui já que será utilizada apenas para validação em dias onde não há publicação
```suggestion
        link_diario = response.css("a[target='diario']::attr(href)")
        data_no_link = link_diario.re_first(r"\d{4}\.\d{2}\.\d{2}")
```